### ISSUE TITLE
from(Class)

### ISSUE THREAD

**Issue by xenoterracide** [dirs-dev/directories-jvm#65]
would it be possible to simply derive the values from a class? for example my personal pattern is `com.xenoteracide.library` (all apps are libraries) which kind of? maps into what's needed for the values needed. I don't have a company, and I'm not certain on that impact so using `xenoterracide` for that would be fine. I'm just thinking this is the general convention for java. Deviance from this can actually cause problems

**Comment by soc** [comment]
@xenoterracide You can just do `ProjectDirectories.from("com.xenoteracide", null, "library"), with the only impact that your Windows paths will lack the nested organization folder.

**Comment by xenoterracide** [comment]
Of course but that doesn't change your API. Anyways it's just an idea. I feel like that naming convention though is the common one for Java. Which is why I'm thinking you could just do it from the class.

**Comment by soc** [comment]
True, I'm just not sure whether it's worth it compared to using the existing API with `ProjectDirectories.from(YourClass.getPackageName(), null, YourClass.getSimpleName())` if that's what you want.

**Comment by xenoterracide** [comment]
Yeah I don't understand the window side of this so much... Like the impact. I would think what it should be is (and I'm on my phone, code typos May abound) "com.xenotericide" "xenotericide" "simple name"  because your organization name is almost always what's after the tld, having just a namespace of com seems weird.

Anyways, it's just an idea for something more conventional. Maybe it's a bad idea, maybe the java convention doesn't work with the mac and Windows conventions.

**Comment by soc** [comment]
As a background info: the function is the way it is (slightly unwieldy), because it tries getting enough information to work on all supported platforms, instead of e. g. having a method for one platform each and putting it on the user to call the right one.
